TheRock CI Workflow Updates 20251106

jayhawk-commits opened the pull request:
- Update the pinned SHA for TheRock in CI workflows.
- Update the version for actions in those same workflows.
- Comment out the rm .patch line and provide details on its use.
- Update manylinux image SHA to match latest.

ScottTodd commented on .github/workflows/therock-ci-linux.yml:
> Comment out the rm .patch line and provide details on its use.

I don't see that change on this PR like you had on https://github.com/ROCm/rocm-systems/commit/524f62ae67a170c6278ad536f0ce275a1492b349. Did you want to include it?

(also, can the `rm` go in the "patch" step or does it have to be in the "fetch sources" step?)

jayhawk-commits replied:
Yes, good call. I will add.

ScottTodd commented on .github/workflows/therock-ci-windows.yml:
Same here, though I see that the "Fetch sources" step is much further down in the file here 🤔

jayhawk-commits commented:
Seeing this failure on the MIOpen build:

```
16.9	In file included from /__w/rocm-libraries/rocm-libraries/projects/miopen/src/graphapi/tensor.cpp:30:
16.9	In file included from /__w/rocm-libraries/rocm-libraries/TheRock/build/third-party/nlohmann-json/dist/include/nlohmann/json.hpp:34:
16.9	In file included from /__w/rocm-libraries/rocm-libraries/TheRock/build/third-party/nlohmann-json/dist/include/nlohmann/adl_serializer.hpp:14:
16.9	/__w/rocm-libraries/rocm-libraries/TheRock/build/third-party/nlohmann-json/dist/include/nlohmann/detail/conversions/from_json.hpp:74:13: error: cannot convert 'const nlohmann::basic_json<>' to 'value_t' without a conversion operator
16.9	   74 |     switch (static_cast<value_t>(j))
16.9	      |             ^~~~~~~~~~~~~~~~~~~~~~~
16.9	/__w/rocm-libraries/rocm-libraries/TheRock/build/third-party/nlohmann-json/dist/include/nlohmann/detail/conversions/from_json.hpp:156:5: note: in instantiation of function template specialization 'nlohmann::detail::get_arithmetic_value<nlohmann::basic_json<>, long, 0>' requested here
16.9	  156 |     get_arithmetic_value(j, val);
16.9	      |     ^
16.9	/__w/rocm-libraries/rocm-libraries/TheRock/build/third-party/nlohmann-json/dist/include/nlohmann/detail/conversions/from_json.hpp:558:16: note: in instantiation of function template specialization 'nlohmann::detail::from_json<nlohmann::basic_json<>>' requested here
16.9	  558 |         return from_json(j, std::forward<T>(val));
16.9	      |                ^
16.9	/__w/rocm-libraries/rocm-libraries/TheRock/build/third-party/nlohmann-json/dist/include/nlohmann/adl_serializer.hpp:31:9: note: in instantiation of function template specialization 'nlohmann::detail::from_json_fn::operator()<nlohmann::basic_json<>, long &>' requested here
16.9	   31 |         ::nlohmann::from_json(std::forward<BasicJsonType>(j), val);
16.9	      |         ^
16.9	/__w/rocm-libraries/rocm-libraries/TheRock/build/third-party/nlohmann-json/dist/include/nlohmann/json.hpp:1828:36: note: in instantiation of function template specialization 'nlohmann::adl_serializer<long>::from_json<const nlohmann::basic_json<> &, long>' requested here
16.9	 1828 |         JSONSerializer<ValueType>::from_json(*this, v);
16.9	      |                                    ^
16.9	/__w/rocm-libraries/rocm-libraries/projects/miopen/src/graphapi/tensor.cpp:317:37: note: in instantiation of function template specialization 'nlohmann::basic_json<>::get_to<long, 0>' requested here
16.9	  317 |     json.at(Tensor::JsonFields::Id).get_to(tensor.mId);
16.9	      |                                     ^
16.9	4 warnings and 1 error generated when compiling for gfx942.
```

jayhawk-commits commented:
`projects/miopen/src/graphapi/` directory has not been touched in months. Same with `nlohmann-json` in TheRock. Has @BrianHarrisonAMD or @DarylHawkinsAMD seen this error before?

BrianHarrisonAMD commented:
> `projects/miopen/src/graphapi/` directory has not been touched in months. Same with `nlohmann-json` in TheRock. Has @BrianHarrisonAMD or @DarylHawkinsAMD seen this error before?

Ah okay I was seeing the same in my branch where I was trying to add MIOpen plugin CI:
- #2309

Trying to figure it out now.

--- 

If I look at a recent build of TheRock where it's working I see this:
```
2025-11-07T13:19:05.6504944Z [442/475] Configure sub-project MIOpen (in background)
2025-11-07T13:19:53.2982885Z [443/475] Configure sub-project rocSPARSE (in background)
2025-11-07T13:22:39.0186280Z [444/475] Building sub-project MIOpen (in background)
2025-11-07T13:22:41.7167600Z [445/475] Stage installing sub-project MIOpen
2025-11-07T13:22:41.8972266Z [446/475] Populate artifact miopen
2025-11-07T13:22:41.9513930Z [447/475] Creating archive /__w/TheRock/TheRock/build/artifacts/miopen_dbg_gfx94X-dcgpu.tar.xz
2025-11-07T13:22:41.9514960Z [448/475] Creating archive /__w/TheRock/TheRock/build/artifacts/miopen_doc_gfx94X-dcgpu.tar.xz
2025-11-07T13:22:41.9515780Z [449/475] Creating archive /__w/TheRock/TheRock/build/artifacts/miopen_dev_gfx94X-dcgpu.tar.xz
2025-11-07T13:22:43.5561739Z [450/475] Creating archive /__w/TheRock/TheRock/build/artifacts/miopen_test_gfx94X-dcgpu.tar.xz
2025-11-07T13:22:43.9132170Z [451/475] Creating archive /__w/TheRock/TheRock/build/artifacts/miopen_run_gfx94X-dcgpu.tar.xz
```

For the rocm-libraries version I see:
```
2025-11-07T00:51:09.8920793Z [292/296] Configure sub-project MIOpen (in background)
2025-11-07T00:52:03.2121033Z [293/296] Building sub-project MIOpen (in background)
2025-11-07T00:52:03.2124362Z FAILED: ml-libs/MIOpen/stamp/build.stamp /__w/rocm-libraries/rocm-libraries/TheRock/build/ml-libs/MIOpen/stamp/build.stamp 
2025-11-07T00:52:03.2127587Z cd /__w/rocm-libraries/rocm-libraries/TheRock/build/ml-libs/MIOpen/build && /opt/python/cp312-cp312/bin/python3.12 /__w/rocm-libraries/rocm-libraries/TheRock/build_tools/teatime.py --log-timestamps --label MIOpen --interactive /__w/rocm-libraries/rocm-libraries/TheRock/build/logs/MIOpen_build.log -- /usr/local/therock-tools/bin/cmake -E env --unset=ROCM_PATH --unset=ROCM_DIR --unset=HIP_PATH --unset=HIP_DIR -- /usr/local/therock-tools/bin/cmake --build /__w/rocm-libraries/rocm-libraries/TheRock/build/ml-libs/MIOpen/build && /usr/local/therock-tools/bin/cmake -E touch /__w/rocm-libraries/rocm-libraries/TheRock/build/ml-libs/MIOpen/stamp/build.stamp
2025-11-07T00:52:03.2200269Z [MIOpen] [0/2] Re-checking globbed directories...
2025-11-07T00:52:03.2201730Z [MIOpen] [1/934] cd /__w/rocm-libraries/rocm-libraries/TheRock/build/ml-libs/MIOpen/build && /usr/local/therock-tools/bin/cmake -E rm -f /__w/rocm-libraries/rocm-libraries/TheRock/build/ml-libs/MIOpen/stamp/stage.stamp
2025-11-07T00:52:03.2202832Z [MIOpen] [2/934] Generating share/miopen/db/gfx900_64.OpenCL.fdb.txt
2025-11-07T00:52:03.2203427Z [MIOpen] [3/934] Generating share/miopen/db/gfx900_56.OpenCL.fdb.txt
2025-11-07T00:52:03.2203884Z [MIOpen] [4/934] Generating share/miopen/db/gfx1030_36.db
```

Looks like there might be some caching happening that is hiding the MIOpen break. 

Trying a fresh build of TheRock to see if I can replicate locally.

--- 

Local TheRock seems fine fresh.
Ill try updating the submodule to see if that is the issue.

---

Okay a fresh local build works.
Seems unrelated to build artifact caching, or any changes to the code.

<img width="2046" height="412" alt="image" src="https://github.com/user-attachments/assets/5107a27e-9f1b-490f-8d79-0a9264d064c2" />

Seems like it's due to how the pre-submit check overrides the directories, and checks things out. 
The CI stage builds in a non-standard way since it specifies a custom CK & custom rocm-libraries to use during pre-submit checks.

jayhawk-commits commented:
Updated to include new rocm-libraries bump from @bsyrowik. CC'ing @bstefanuk as well.

Removed the CK SHA as it was the potential cause for the build issues on MIOpen.

jayhawk-commits commented:
Removing the CK pinned SHA to make it behave as before shows a build error is from CK:
```
2025-11-07T23:13:36.8330560Z [composable_kernel] Exception Code: 0xC0000005
2025-11-07T23:13:36.8331107Z [composable_kernel]  #0 0x00007ff63f1b49b8 (C:\A3B1FD1A-CAB8-4F1A-B846-94A0E4C652E5\build\core\clr\dist\lib\llvm\bin\clang++.exe+0x1b249b8)
2025-11-07T23:13:36.8332044Z [composable_kernel]  #1 0x00007ff63f18ad04 (C:\A3B1FD1A-CAB8-4F1A-B846-94A0E4C652E5\build\core\clr\dist\lib\llvm\bin\clang++.exe+0x1afad04)
2025-11-07T23:13:36.8332868Z [composable_kernel]  #2 0x00007ff63fb569b5 (C:\A3B1FD1A-CAB8-4F1A-B846-94A0E4C652E5\build\core\clr\dist\lib\llvm\bin\clang++.exe+0x24c69b5)
2025-11-07T23:13:36.8333771Z [composable_kernel]  #3 0x00007ff63f40ca4c (C:\A3B1FD1A-CAB8-4F1A-B846-94A0E4C652E5\build\core\clr\dist\lib\llvm\bin\clang++.exe+0x1d7ca4c)
2025-11-07T23:13:36.8334595Z [composable_kernel]  #4 0x00007ff63f189ceb (C:\A3B1FD1A-CAB8-4F1A-B846-94A0E4C652E5\build\core\clr\dist\lib\llvm\bin\clang++.exe+0x1af9ceb)
2025-11-07T23:13:36.8335489Z [composable_kernel]  #5 0x00007ff63f40c8c4 (C:\A3B1FD1A-CAB8-4F1A-B846-94A0E4C652E5\build\core\clr\dist\lib\llvm\bin\clang++.exe+0x1d7c8c4)
2025-11-07T23:13:36.8336314Z [composable_kernel]  #6 0x00007ff63f3f607d (C:\A3B1FD1A-CAB8-4F1A-B846-94A0E4C652E5\build\core\clr\dist\lib\llvm\bin\clang++.exe+0x1d6607d)
2025-11-07T23:13:36.8337112Z [composable_kernel]  #7 0x00007ff63f494620 (C:\A3B1FD1A-CAB8-4F1A-B846-94A0E4C652E5\build\core\clr\dist\lib\llvm\bin\clang++.exe+0x1e04620)
2025-11-07T23:13:36.8337977Z [composable_kernel]  #8 0x00007ff63d74dfb0 (C:\A3B1FD1A-CAB8-4F1A-B846-94A0E4C652E5\build\core\clr\dist\lib\llvm\bin\clang++.exe+0xbdfb0)
2025-11-07T23:13:36.8338792Z [composable_kernel]  #9 0x00007ff63d747f98 (C:\A3B1FD1A-CAB8-4F1A-B846-94A0E4C652E5\build\core\clr\dist\lib\llvm\bin\clang++.exe+0xb7f98)
2025-11-07T23:13:36.8339795Z [composable_kernel] #10 0x00007ff63d749327 (C:\A3B1FD1A-CAB8-4F1A-B846-94A0E4C652E5\build\core\clr\dist\lib\llvm\bin\clang++.exe+0xb9327)
2025-11-07T23:13:36.8340582Z [composable_kernel] #11 0x00007ff63d757514 (C:\A3B1FD1A-CAB8-4F1A-B846-94A0E4C652E5\build\core\clr\dist\lib\llvm\bin\clang++.exe+0xc7514)
2025-11-07T23:13:36.8341374Z [composable_kernel] #12 0x00007ff640c05f78 (C:\A3B1FD1A-CAB8-4F1A-B846-94A0E4C652E5\build\core\clr\dist\lib\llvm\bin\clang++.exe+0x3575f78)
2025-11-07T23:13:36.8342052Z [composable_kernel] #13 0x00007ff8e8e34cb0 (C:\Windows\System32\KERNEL32.DLL+0x14cb0)
2025-11-07T23:13:36.8342610Z [composable_kernel] #14 0x00007ff91c77edcb (C:\Windows\SYSTEM32\ntdll.dll+0x7edcb)
2025-11-07T23:13:36.8343147Z [composable_kernel] clang++: error: clang frontend command failed due to signal (use -v to see invocation)
2025-11-07T23:13:36.8344185Z [composable_kernel] AMD clang version 22.0.0git (https://github.com/ROCm/llvm-project.git cf5a4bf00b403b30663fb49a1ac0d2d8add48356+PATCHED:93c451b46cc0dc23c47d67e394b370de65731aac)
2025-11-07T23:13:36.8344864Z 
2025-11-07T23:13:36.8344985Z [composable_kernel] Target: x86_64-pc-windows-msvc
2025-11-07T23:13:36.8345330Z 
2025-11-07T23:13:36.8345412Z [composable_kernel] Thread model: posix
2025-11-07T23:13:36.8345570Z 
2025-11-07T23:13:36.8345860Z [composable_kernel] InstalledDir: C:\A3B1FD1A-CAB8-4F1A-B846-94A0E4C652E5\build\core\clr\dist\lib\llvm\bin
2025-11-07T23:13:36.8346225Z 
2025-11-07T23:13:36.8346486Z [composable_kernel] clang++: note: diagnostic msg: Error generating preprocessed source(s).
```

jayhawk-commits commented:
@BrianHarrisonAMD the compiler has had some bumps on TheRock recently. The issue could be coming from changes there?
